scripts/release.py

```diff
@@ -0,0 +1,199 @@
+            tag = "v%s" % name
+            branch = "1.x"
+        # if this is the rc+1 for this base
+        else:
+            name = "%s.0rc%s" % (base, str(rc_version + 1))
+            tag = "v%s" % name
+            branch = base
+        rn_raw = generate_rn(branch)
+        rn = clean_rn(rn_raw)
+        create_draft_release(branch=branch, name=name, tag=tag, dd_repo=dd_repo, rn=rn, prerelease=True)
+
+    # patch release
+    elif patch:
+        # figure out the patch version we want
+        search = "v%s.((\d+))" % base
+        tags = dd_repo.get_tags()
+        patch_version = 1
+        for tag in tags:
+            try:
+                patch_num = re.findall(search, tag.name)[0][0]
+                patch_num = int(patch_num)
+            except (IndexError, ValueError, TypeError):
+                continue
+            if patch_num > patch_version:
+                patch_version = patch_num
+
+        name = "%s.%s" % (base, str(patch_version + 1))
+        tag = "v%s" % name
+        rn_raw = generate_rn(base)
+        rn = clean_rn(rn_raw)
+        create_draft_release(branch=base, name=name, tag=tag, dd_repo=dd_repo, rn=rn, prerelease=False)
+
+    # minor release
+    else:
+        name = "%s.0" % base
+        tag = "v%s" % name
+        branch = base
+        rn_raw = generate_rn(branch)
+        # get anything in unreleased section in case there were updates since the last RC
+        unreleased = (
+            rn_raw.decode().split("## v")[0].replace("\n## Unreleased\n", "", 1).replace("# Release Notes\n", "", 1)
+        )
+        unreleased = unreleased.split("###")
+        try:
+            unreleased_sections = dict(section.split("\n\n-") for section in unreleased)
+        except ValueError:
+            unreleased_sections = {}
+        relevant_rns = []
+        if unreleased_sections:
+            relevant_rns.append(unreleased_sections)
+
+        rns = rn_raw.decode().split("## v")
+        split_str = "## v%s" % base
+        for rn in rns:
+            if rn.startswith("%s.0" % base):
+                # cut out the version section
+                sections = rn.split("###")[1:]
+                prelude_section = {}
+                # if there is a prelude, we need to grab that separately since it has different syntax
+                if sections[0].startswith(" Prelude\n\n"):
+                    prelude_section[" Prelude"] = sections[0].split("\n\n", 1)[1]
+                    sections = sections[1:]
+                sections_dict = {**dict(section.split("\n\n", 1) for section in sections), **prelude_section}
+                relevant_rns.append(sections_dict)
+        # join all the sections from different relevant RCs together
+        keys = set().union(*relevant_rns)
+        rns_dict = {k: "".join(dic.get(k, "") for dic in relevant_rns) for k in keys}
+        rns_dict_clean = {}
+        for key in rns_dict.keys():
+            rns_dict_clean[key.lstrip()] = rns_dict[key]
```

Comment:
Same here (extract into helper to simplify logic here). My main concern is that someone without context (for example, me) of how the PyGithub library works might be overwhelmed by what's going on, and extracting this into a helper might make more sense to readers in the future.